File: PRAK601-2310817120012-RirinCitraLestari.c
#include <stdio.h>

void createMatrix(int rows, int cols, int values[]) {
    int matrix[rows][cols];

    int k = 0;
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            matrix[i][j] = values[k++];
        }
    }

    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            printf("%d ", matrix[i][j]);
        }
        printf("\n");
    }
}

int main() {
    int rows, cols;
    scanf("%d %d", &rows, &cols);

    int size = rows * cols;
    int values[size];

    for (int i = 0; i < size; i++) {
        scanf("%d", &values[i]);
    }

    createMatrix(rows, cols, values);

    return 0;
}

File: PRAK601-2310817120012-RirinCitraLestari.py
def create_matrix(rows, cols, values):
    matrix = [values[i * cols:(i + 1) * cols] for i in range(rows)]
    return matrix

def main():
    rows, cols = map(int, input().split())
    values = list(map(int, input().split()))
    matrix = create_matrix(rows, cols, values)

    for row in matrix:
        print(*row)

if __name__ == "__main__":
    main()
